python/dolfinx/fem/element.py

```diff
@@ -160,3 +162,190 @@ def _(e: basix.finite_element.FiniteElement):
+
+    @property
+    def interpolation_ident(self) -> bool:
+        """Check if interpolation into the finite element space is an identity operation given the
+        evaluation on an expression at specific points, i.e. the degree-of-freedom are equal to
+        point evaluations. The function will return `true` for Lagrange elements."""
+        return self._cpp_object.interpolation_ident
+
+    @property
+    def space_dimension(self) -> int:
+        """Dimension of the finite element function space (the number of degrees-of-freedom for the
+        element).
+
+        For 'blocked' elements, this function returns the dimension of the full element rather than
+        the dimension of the base element.
+        """
+        return self._cpp_object.space_dimension
+
+    @property
+    def needs_dof_transformations(self) -> bool:
+        """Check if DOF transformations are needed for this element.
+
+        DOF transformations will be needed for elements which might not be continuous when two
+        neighbouring cells disagree on the orientation of a shared sub-entity, and when this cannot
+        be corrected for by permuting the DOF numbering in the dofmap.
+
+        For example, Raviart-Thomas elements will need DOF transformations, as the neighbouring
+        cells may disagree on the orientation of a basis function, and this orientation cannot be
+        corrected for by permuting the DOF numbers on each cell.
+        """
+        return self._cpp_object.needs_dof_transformations
+
+    @property
+    def signature(self) -> str:
+        """String identifying the finite element."""
+        return self._cpp_object.signature
+
+    def T_apply(self, x: npt.NDArray[np.floating], cell_permutations: np.int32, dim: int) -> None:
+        """Transform basis functions from the reference element ordering and orientation to the
+        globally consistent physical element ordering and orientation.
+
+        Args:
+            x: Data to transform (in place). The shape is ``(m, n)``, where `m` is the number of
+            dgerees-of-freedom and the storage is row-major.
+            cell_permutations: Permutation data for the cell.
+            dim: Number of columns in ``data``.
+
+        Note:
+            Exposed for testing. Function is not vectorised across multiple cells. Please see
+            `basix.numba_helpers` for performant versions.
+        """
+        self._cpp_object.T_apply(x, cell_permutations, dim)
+
+    def Tt_apply(self, x: npt.NDArray[np.floating], cell_permutations: np.int32, dim: int) -> None:
+        """Apply the transpose of the operator applied by T_apply().
+
+        Args:
+            x: Data to transform (in place). The shape is ``(m, n)``, where `m` is the number of
+            dgerees-of-freedom and the storage is row-major.
+            cell_permutations: Permutation data for the cell.
+            dim: Number of columns in `data`.
+
+        Note:
```

Comment:
@schnellerhase  This is not correct. These functions have loop access in the python wrapper.

Reply:
The type hint should be `npt.NDArray[np.int32]`.
https://github.com/FEniCS/dolfinx/blob/fcc7719793b55da5f3736a73e109c76b114bdf8d/python/dolfinx/wrappers/fem.cpp#L187-L196

Reply:
Thanks for the catches and fixes - my bad!